modules/services/tailscale-systray.nix

```diff
@@ -0,0 +1,41 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.tailscale-systray;
+in
+{
+  meta.maintainers = [ lib.maintainers.yethal ];
+
+  options.services.tailscale-systray = {
+    enable = lib.mkEnableOption "Official Tailscale systray application for Linux";
+
+    package = lib.mkPackageOption pkgs "tailscale" { };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      (lib.hm.assertions.assertPlatform "services.tailscale" pkgs lib.platforms.linux)
+    ];
+
+    systemd.user.services.tailscale-systray = {
+      Unit = {
+        Description = "Official Tailscale systray application for Linux";
+        Requires = [ "tray.target" ];
+        After = [
+          "graphical-session.target"
+          "tray.target"
+        ];
+        PartOf = [ "graphical-session.target" ];
+      };
+      Install = {
+        WantedBy = [ "graphical-session.target" ];
+      };
+      Service.ExecStart = "${lib.getExe cfg.package} systray";
```

Comment:
Interesting... have I been doing this wrong and using `tailscale-systray` instead of `tailscale systray` lol

Reply:
Hmm. that actually complicates things more.. The module would be unusable for user's getting tailscale from nixpkgs. Seems like we should default to using `tailscale-systray` until the version in nixpkgs supports that argument, no?

Reply:
Version bump was merged two days ago, no idea how long until it hits unstable though.

The problem with using `tailscale-systray` package here is that it's a third-party package that's not affiliated with tailscale in any way so substituting it as the official systray app seems fishy. If we wanted to support both then maybe having two modules, `tailscale-systray` and `tailscale-systray-official` would be preferred

Reply:
Oh, that's going through staging... https://github.com/NixOS/nixpkgs/pull/442245 Yeah, that will take a bit.

Reply:
Could add an assertion that checks the pkg version configured. Then the module won't let you accidentally enable it on the wrong version of the package.